### PR TITLE
test-bot: Install dependencies for Cask container tests.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

In preparation of https://github.com/Homebrew/brew/pull/725, 

- `adobe-air`
- `cabextract`
- `unar` and
- `xz`

have to be installed before running `brew cask-tests`, otherwise some of the container tests will be skipped.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> have to be installed before running brew cask-tests, otherwise some of the container tests will be skipped.

Longer-term we need to figure out a way of testing these things that doesn't require additional dependencies being installed on our bots (like `brew tests` does). Eventually we'll want to merge `brew tests` and `brew cask-tests` too.

Unfortunately the `brew install` does an implicit tap of homebrew/core which will be relatively slow. Where are the casks stored in this case? It's not testing a `/usr/local` Homebrew install.

**Comment by reitermarkus** [comment]
> Longer-term we need to figure out a way of testing these things that doesn't require additional dependencies

I completely agree, however, I don't see a way to test the correct detection and extraction of containers without having the tools to extract them installed. I think we can neglect at least `cabextract` and `xz` for now, as there aren't any Casks using them. Also only one using `unar` will be left after https://github.com/caskroom/homebrew-cask/pull/23918 is merge.

**Comment by MikeMcQuaid** [comment]
> I completely agree, however, I don't see a way to test the correct detection and extraction of containers without having the tools to extract them installed. I think we can neglect at least cabextract and xz for now, as there aren't any Casks using them.

Sounds like a good argument to 💀 that code 😉.

File: Library/Homebrew/dev-cmd/test-bot.rb [hunk 1]
```diff
@@ -659,7 +659,13 @@ def homebrew
         test "brew", "readall", "--syntax"
         if OS.mac? &&
            (HOMEBREW_REPOSITORY/"Library/Homebrew/cask/cmd/brew-cask-tests.rb").exist?
-          run_as_not_developer { test "brew", "tap", "caskroom/cask" }
+          run_as_not_developer do
+            test "brew", "tap", "caskroom/cask"
+            test "brew", "cask", "install", "https://raw.githubusercontent.com/caskroom/homebrew-cask/master/Casks/adobe-air.rb"
+            test "brew", "install", "https://raw.githubusercontent.com/Homebrew/homebrew-core/master/Formula/cabextract.rb"
+            test "brew", "install", "https://raw.githubusercontent.com/Homebrew/homebrew-core/master/Formula/unar.rb"
+            test "brew", "install", "https://raw.githubusercontent.com/Homebrew/homebrew-core/master/Formula/xz.rb"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/test-bot.rb]: Would it be enough to have these installed systemwide on the build machines? This means they would likely not be kept particularly up-to-date but might be nicer than sticking them here.

Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/test-bot.rb]: If that's possible, yes, as long as they can extract the containers I don't care if they are up-to-date.

**Comment by ilovezfs** [comment]
:-1:

**Comment by MikeMcQuaid** [comment]
Passing on this because I'd like to get `brew cask-tests` so it doesn't require `caskroom` to be tapped and this is going in the opposite direction.

**Pull request closed without merging**